File: src/main/java/com/aaronhowser1/lofirecordstomineto/LoFiRecordsToMineTo.java
package com.aaronhowser1.lofirecordstomineto;

import net.minecraft.world.item.CreativeModeTabs;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.BuildCreativeModeTabContentsEvent;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

@Mod(LoFiRecordsToMineTo.MOD_ID)
public class LoFiRecordsToMineTo {
    public static final String MOD_ID = "lofirecordstomineto";

    public LoFiRecordsToMineTo() {
        IEventBus eventBus = FMLJavaModLoadingContext.get().getModEventBus();

        ModItems.ITEMS_REGISTRY.register(eventBus);
        ModSounds.SOUND_EVENT_REGISTRY.register(eventBus);

        MinecraftForge.EVENT_BUS.register(this);

        eventBus.addListener(LoFiRecordsToMineTo::addToCreativeTab);
    }

    private static void addToCreativeTab(BuildCreativeModeTabContentsEvent event) {
        if (event.getTabKey() != CreativeModeTabs.TOOLS_AND_UTILITIES) return;

        event.acceptAll(
                ModItems.ITEMS_REGISTRY
                        .getEntries()
                        .stream()
                        .map((entry) -> entry.get().getDefaultInstance())
                        .toList()
        );
    }

}

File: src/main/kotlin/aaronhowser/mods/lofirecordstomineto/LoFiRecordsToMineTo.kt
package aaronhowser.mods.lofirecordstomineto

import aaronhowser.mods.lofirecordstomineto.registry.ModItems
import aaronhowser.mods.lofirecordstomineto.registry.ModSounds
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.item.CreativeModeTabs
import net.neoforged.fml.common.Mod
import net.neoforged.neoforge.event.BuildCreativeModeTabContentsEvent
import thedarkcolour.kotlinforforge.neoforge.forge.MOD_BUS

@Mod(LoFiRecordsToMineTo.ID)
object LoFiRecordsToMineTo {
    const val ID = "lofirecordstomineto"

    fun getRl(string: String): ResourceLocation = ResourceLocation.fromNamespaceAndPath(ID, string)

    init {
        ModItems.ITEM_REGISTRY.register(MOD_BUS)
        ModSounds.SOUND_EVENT_REGISTRY.register(MOD_BUS)

        MOD_BUS.addListener(::addToCreativeTab)
    }

    private fun addToCreativeTab(event: BuildCreativeModeTabContentsEvent) {
        if (event.tabKey != CreativeModeTabs.TOOLS_AND_UTILITIES) return

        val items = ModItems.ITEM_REGISTRY.entries.map { it.get().defaultInstance }
        event.acceptAll(items)
    }

}